templates/webhook.bin.erb

```diff
@@ -250,6 +259,24 @@ class Server < Sinatra::Base
       message
     end
 
+    def generate_types(environment)
+      begin
+        command = "#{$command_prefix} /opt/puppetlabs/bin/puppet generate types --environment #{environment}"
+
+        message = run_command(command)
+        $logger.info("message: #{message} environment: #{environment}")
+        status_message = {:status => :success, :message => message.to_s, :environment => environment, :status_code => 200}
+        notify_slack(status_message) if slack?
+        status_message.to_json
```

Comment:
where does this go?  I think in deploy it's the json that ends up being returned to the http client.

Reply:
Good point, I removed the status_message.to_json line.

I left the rest of the code there. We want the logging to happen still and we want to be able to send notifications to slack if generate_types succeeds or fails.